### PR TITLE
feat: [CHAOS-2814]: {NPCI}: Alerts Feature Request

### PR DESCRIPTION
**Pull request by vanshBhatia-A4k9**
## Describe your changes

- Registered 1 new notification entity `CHAOS_EXPERIMENT`.
- Registered 3 notification events for the entity `CHAOS_EXPERIMENT_STARTED`, `CHAOS_EXPERIMENT_COMPLETED`, and `CHAOS_EXPERIMENT_STOPPED`.
- Registered all the relevant templates for the above entity-event pair in the notificationTemplates collection as part of NG-Manager start time calls:

Screenshot 2023-11-30 at 5 11 08 PM

- Added a new internal API that will be used to trigger all chaos experiment related events

Sample email test:

Screenshot 2023-11-30 at 5 06 22 PM

### Notification Rule required/used to test the changes locally:

```
{
  "_id": "65650a4a6b10870759d04918",
  "identifier": "NR1",
  "accountIdentifier": "kmpySmUISimoRrJL6NL73w",
  "notificationConditions": [
    {
      "conditionName": "del-condition",
      "notificationEventConfigs": [
        {
          "notificationEntity": "CHAOS_EXPERIMENT",
          "notificationEvent": "CHAOS_EXPERIMENT_STARTED",
          "notificationEventParameters": {
            "delegateGroupIdentifiers": [],
            "_class": "io.harness.notification.entities.eventmetadata.DelegateNotificationEventParameters"
          },
          "notificationChannels": [
            {
              "identifier": "nr1",
              "notificationChannelType": "EMAIL",
              "status": "ENABLED",
              "channel": {
                "emailIds": [
                  "vansh.bhatia@harness.io"
                ],
                "_class": "io.harness.notification.entities.EmailChannel"
              },
              "nextIteration": 0
            }
          ]
        },
        {
          "notificationEntity": "CHAOS_EXPERIMENT",
          "notificationEvent": "CHAOS_EXPERIMENT_STOPPED",
          "notificationEventParameters": {
            "delegateGroupIdentifiers": [],
            "_class": "io.harness.notification.entities.eventmetadata.DelegateNotificationEventParameters"
          },
          "notificationChannels": [
            {
              "identifier": "nr1",
              "notificationChannelType": "EMAIL",
              "status": "ENABLED",
              "channel": {
                "emailIds": [
                  "vansh.bhatia@harness.io"
                ],
                "_class": "io.harness.notification.entities.EmailChannel"
              },
              "nextIteration": 0
            }
          ]
        },
        {
          "notificationEntity": "CHAOS_EXPERIMENT",
          "notificationEvent": "CHAOS_EXPERIMENT_COMPLETED",
          "notificationEventParameters": {
            "delegateGroupIdentifiers": [],
            "_class": "io.harness.notification.entities.eventmetadata.DelegateNotificationEventParameters"
          },
          "notificationChannels": [
            {
              "identifier": "nr1",
              "notificationChannelType": "EMAIL",
              "status": "ENABLED",
              "channel": {
                "emailIds": [
                  "vansh.bhatia@harness.io"
                ],
                "_class": "io.harness.notification.entities.EmailChannel"
              },
              "nextIteration": 0
            }
          ]
        }
      ]
    }
  ],
  "nextIteration": 0,
  "_class": "notificationRule",
  "orgIdentifier": "default",
  "projectIdentifier": "testing"
}
```

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/56169)

### CODE REVIEW

**Comment by vanshBhatia-A4k9** [comment]
trigger uts

File: 950-delegate-tasks-beans/src/main/proto/io/harness/notification/notification_request.proto [hunk 1]
```diff
@@ -17,6 +17,7 @@ enum Team {
   PL = 6;
   GTM = 7;
   IDP = 8;
+  CHAOS = 9;
```
Comment (jennyjames21) [inline 950-delegate-tasks-beans/src/main/proto/io/harness/notification/notification_request.proto:20]: You can remove this change for now, will add team later if needed

**Review by jennyjames21 — APPROVED** [review]

**Review by amityt — APPROVED** [review]

**Comment by vanshBhatia-A4k9** [comment]
trigger uts

**Comment by vanshBhatia-A4k9** [comment]
trigger ut1

**Comment by vanshBhatia-A4k9** [comment]
trigger ut4

**Comment by vanshBhatia-A4k9** [comment]
trigger uts

**Comment by vanshBhatia-A4k9** [comment]
trigger ut4

**Pull request merged**